aisdc/attacks/likelihood_attack.py

```diff
@@ -80,14 +79,20 @@ def __init__(self, **kwargs):
         self.__dict__["n_shadow_rows_confidences_min"] = 10
         self.__dict__["p_thresh"] = 0.05
         self.__dict__["report_name"] = None
-        self.__dict__["attack_config_json_file_name"] = "config.json"
+        self.__dict__["training_data_filename"] = None
+        self.__dict__["test_data_filename"] = None
+        self.__dict__["training_preds_filename"] = None
+        self.__dict__["test_preds_filename"] = None
+        self.__dict__["target_model"] = None
+        self.__dict__["target_model_hyp"] = None
+        self.__dict__["attack_config_json_file_name"] = None
         self.__dict__["shadow_models_fail_fast"] = False
-
-        if os.path.isfile(self.__dict__["attack_config_json_file_name"]):
-            self.load_config_file_into_dict(
-                self.__dict__["attack_config_json_file_name"]
-            )
         self.__dict__.update(kwargs)
+        configfile_obj = ConfigFile(self.__dict__["attack_config_json_file_name"])
+        dict_obj = configfile_obj.load_config_file_into_dict(self.__dict__)
```

Comment:
won't `self.__dict__` be passed by reference, so the things you add don't then need to be added again to `self.__dict__` with the update statement on line 93?

Reply:
Thanks, I agree and changes made as suggested.